### PR TITLE
refactor: Remove ICustomClippingElement from Skia/Wasm

### PR DESCRIPTION
**Pull request by Youssef1313**
Skia and Wasm now have proper clipping logic, and for the most part, `ICustomClippingElement` should not be needed.

If a specific control **really** needs to control clipping (e.g, Canvas), it should override `GetLayoutClip` method that was recently added.

## PR Type

What kind of change does this PR introduce?

### <samp>🤖 Generated by Copilot at e3ae662</samp>

This pull request removes the `ICustomClippingElement` interface from various UI classes and refactors the clipping logic to use conditional compilation symbols. This simplifies the cross-runtime compatibility and prepares for the migration to WinUI. The interface is only implemented by the `FrameworkElement` class and its subclasses when the `__CROSSRUNTIME__` or `HAS_UNO_WINUI` symbols are not defined. The mock and reference files are also updated accordingly.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/ICustomClippingElement.cs [hunk 1]
```diff
@@ -1,22 +1,23 @@
-﻿namespace Windows.UI.Xaml
+﻿#if !__CROSSRUNTIME__ && !IS_UNIT_TESTS
+namespace Windows.UI.Xaml;
+
+internal interface ICustomClippingElement
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/ICustomClippingElement.cs:4]: This interface is now only needed for platforms that don't have the proper clipping logic. So removing it from Skia/Wasm where this was fixed in https://github.com/unoplatform/uno/pull/13695, and also removing from unit tests where it's not really relevant at all as unit tests won't notice the effect of this. Any clipping test should be either a runtime test or a UI test.

**Review by dr1rrb — APPROVED** [review]
LGTM, really nice that we are finally able to remove that patch interface :)

**Comment by jeromelaban** [comment]
@Youssef1313 did you check the screenshot diffs?

**Review by carldebilly — APPROVED** [review]

**Comment by Youssef1313** [comment]
@jeromelaban I reviewed the screenshot comparison. Nothing there regressed.

**Review by ajpinedam — APPROVED** [review]

**Comment by Youssef1313** [comment]
@jeromelaban Is this good to merge?

**Pull request merged**